Python/ceval_gil.c

```diff
@@ -463,63 +464,99 @@ take_gil(PyThreadState *tstate)
 
 void _PyEval_SetSwitchInterval(unsigned long microseconds)
 {
-    struct _gil_runtime_state *gil = &_PyRuntime.ceval.gil;
+    PyInterpreterState *interp = _PyInterpreterState_Get();
+    struct _gil_runtime_state *gil = interp->ceval.gil;
+    assert(gil != NULL);
     gil->interval = microseconds;
```

Comment:
Do we need some way to propagate this to new subinterpreters? I don't see it if it's here.

Not sure whether this would catch people out or not. Presumably increasing the switch count is less important with proper parallelism, and it's documented as setting "the interpreter's switch interval" so we're clear on that front. But it may be surprising.

Probably only worth adding a note to the sys docs to clarify that it doesn't propagate to new subinterpreter GILs.

Reply:
Yeah, we'll make sure the docs are clear.

Reply:
I've left a TODO note on the issue: https://github.com/python/cpython/issues/99113#issuecomment-1536687011.